config/config.go

```diff
@@ -921,17 +922,32 @@ func fetchConfig(u *url.URL) ([]byte, error) {
 	}
 	req.Header.Add("Accept", "application/toml")
 	req.Header.Set("User-Agent", internal.ProductToken())
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return nil, err
-	}
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to retrieve remote config: %s", resp.Status)
+	retries := 3
+	for i := 1; i <= retries; i++ {
```

Comment:
if retries is 0, this won't attempt at all. If we later move to user-configurable retries, that won't quite make sense

Reply:
Right changed i to 0